rllib/models/catalog.py

```diff
@@ -770,22 +776,10 @@ def get_preprocessor_for_space(
                     )
                 )
 
-        if options.get("custom_preprocessor"):
```

Comment:
As per discussion with @sven1977 , I'm fully deprecating custom_preprocessors in this PR.
This method is @DeveloperAPI, so I've moved the actual deprecation error to algorithm config.